subprojects/dependency-management/src/main/java/org/gradle/internal/resource/transfer/DefaultCacheAwareExternalResourceAccessor.java

```diff
@@ -115,7 +115,8 @@ public ExternalResourceMetaData create() {
 
                     if (isUnchanged) {
                         LOGGER.info("Cached resource {} is up-to-date (lastModified: {}).", location, cached.getExternalLastModified());
-                        // TODO - update the index with the new remote meta-data
+                        // Update the cache entry in the index: this resets the age of the cached entry to zero
+                        cachedExternalResourceIndex.store(location.toString(), cached.getCachedFile(), cached.getExternalResourceMetaData());
```

Comment:
The call to `store` is ultimately guarded under a `cacheLockingManager`, so I'm not sure if it's necessary or beneficial to explicitly lock the cache here. https://github.com/gradle/gradle/blob/b4c9e19e0da352ac80be67805e7539c426cb13e2/subprojects/dependency-management/src/main/java/org/gradle/internal/resource/cached/ivy/AbstractCachedIndex.java#L72-L78